phys2denoise.py

```diff
@@ -0,0 +1,234 @@
+
+    Notes
+    -----
+    The code was greatly copied from phys2bids (copyright the physiopy community)
+
+    """
+    # Check options to make them internally coherent pt. I
+    # #!# This can probably be done while parsing?
+    outdir = os.path.abspath(outdir)
+    os.makedirs(outdir)
+    os.makedirs(os.path.join(outdir, 'code'))
+    log_path = os.path.join(outdir, 'code', 'logs')
+    os.makedirs(log_path)
+
+    # Create logfile name
+    basename = 'phys2denoise_'
+    extension = 'tsv'
+    isotime = datetime.datetime.now().strftime('%Y-%m-%dT%H%M%S')
+    logname = os.path.join(log_path, (basename + isotime + '.' + extension))
+
+    # Set logging format
+    log_formatter = logging.Formatter(
+        '%(asctime)s\t%(name)-12s\t%(levelname)-8s\t%(message)s',
+        datefmt='%Y-%m-%dT%H:%M:%S')
+
+    # Set up logging file and open it for writing
+    log_handler = logging.FileHandler(logname)
+    log_handler.setFormatter(log_formatter)
+    sh = logging.StreamHandler()
+
+    if quiet:
+        logging.basicConfig(level=logging.WARNING,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+    elif debug:
+        logging.basicConfig(level=logging.DEBUG,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+    else:
+        logging.basicConfig(level=logging.INFO,
+                            handlers=[log_handler, sh],
+                            format='%(levelname)-10s %(message)s')
+
+    version_number = _version.get_versions()['version']
+    LGR.info(f'Currently running phys2denoise version {version_number}')
+    LGR.info(f'Input file is {filename}')
+
+    # Save call.sh
+    arg_str = ' '.join(sys.argv[1:])
+    call_str = f'phys2denoise {arg_str}'
+    f = open(os.path.join(log_path, 'call.sh'), "a")
+    f.write(f'#!bin/bash \n{call_str}')
+    f.close()
+
+    # Check options to make them internally coherent pt. II
+    # #!# This can probably be done while parsing?
+    # filename, ftype = utils.check_input_type(filename)
+
+    if not os.path.isfile(filename) and filename is not None:
+        raise FileNotFoundError(f'The file {filename} does not exist!')
+
+    # Read input file
+    physio = np.genfromtxt(filename)
+
+    # Prepare pandas dataset
+    regr = pd.DataFrame()
+
+    # Goes through the list of metrics and calls them
+    for metric in metrics:
+        if metrics == 'retroicor_card':
+            args = select_input_args(compute_retroicor_regressors, metric_args)
+            args['card'] = True
+            print_metric_call(metric, args)
+            regr['retroicor_card'] = compute_retroicor_regressors(physio,
+                                                                  **args)
+        elif metrics == 'retroicor_resp':
+            args = select_input_args(compute_retroicor_regressors, metric_args)
+            args['resp'] = True
```

Comment:
Doesn't make more sense to modify [this](https://github.com/CesarCaballeroGaudes/phys2denoise/blob/4ad18ba408801976ec0373ff813252499106b72d/phys2denoise/metrics/retroicor.py#L74) so we don't have two if conditions, one here and another one in the retroicor code?

Reply:
What do you mean @vinferrer ? The two `if`s do different things (Or you mean to recode the three functions in retroicor intotwo? But then there's quite a lot of lines that are common...)

Reply:
Okay, after seeing that you don't input the metrics argument directly, why don't you put this line inside ``select_input_args`` I think you can reduce this if statement at least, another thing is that  instead of creating the dictionary entry ``args['card'] = True`` when ``metrics == 'retroicor_card'`` you could use directly the ``metrics == 'retroicor_card'`` inside ``compute_retroicor_regressors``. Do you follow? or should I do suggestions in both PR's?

Reply:
I think we can avoid an extra if statement

Reply:
I mean, you have a function called ``select_input_args`` I would include the type of metric detection inside of this function rather than in the main workflow